version.py

```diff
@@ -117,3 +120,12 @@ def find_latest_version_with_matching_major(reference_version: semver.VersionInf
             if not latest_candidate or latest_candidate < candidate:
                 latest_candidate = candidate
     return latest_candidate
+
+
+def partition_by_major_and_minor(
```

Comment:
a _brief_ docstr would be appropriate, I suppose
also, consider sorting the versions (maybe use order set..)